### PR TITLE
Update CMake code and libaec

### PR DESCRIPTION
**Pull request by byrnHDF**
Update to latest hdf standard usage.

### CODE REVIEW

File: config/cmake/README.txt.cmake.in [hunk 1]
```diff
@@ -58,7 +67,7 @@ To test the installation with the examples;
 
 When executed, the ctest script will save the results to the log file, test.log, as
 indicated by the ctest command. If you wish the to see more build and test information,
-add "-VV" to the ctest command. The output should show near the end;
+add "-VV" to the ctest command. The output should show;
```
Comment (bmribler) [inline config/cmake/README.txt.cmake.in:70]: Is ';' supposed to be ':'?

File: release_notes/RELEASE.txt [hunk 2]
```diff
@@ -38,6 +38,23 @@ New features and changes
 ========================
     Configuration:
     -------------
+    - Corrected path searched by CMake find_package command
+
+      The install path for cmake find_package files had been changed to use
+        "share/cmake"
+      for all platforms. However the trailing "hdf4" directory was not removed.
```
Comment (bmribler) [inline release_notes/RELEASE.txt:45]: Maybe: "for all platforms, but the trailing "hdf4" was left by mistake."?
Also, I wonder something like this for the first sentence:
"The install path for cmake find_package files had been changed to use "share/cmake" in version... for all platforms."  More clear, I think.  Of course, take my English with caution! :D

File: release_notes/RELEASE.txt [hunk 3]
```diff
@@ -38,6 +38,23 @@ New features and changes
...
+      This "hdf4" additional directory has been removed.
+
+      (ADB - 2021/09/27)
+
+    - Corrected pkg-config compile script
+
+      It was discovered that the position of the "$@" argument for the command
+      in the compile script may fail on some platforms and configurations. The
+      position of the "$@"command argument was moved before the pkg-config sub command.
```
Comment (bmribler) [inline release_notes/RELEASE.txt:54]: "...was moved to before..."?

**Review by bmribler — APPROVED** [review]

File: config/cmake/README.txt.cmake.in (hunk 1 above)
Comment (byrnHDF) [reply config/cmake/README.txt.cmake.in:70]: It's been ";" forever, so I think it will be okay.

File: release_notes/RELEASE.txt (hunk 3 above)
Comment (byrnHDF) [reply release_notes/RELEASE.txt:54]: yes.

**Review by lrknox — APPROVED** [review]

**Pull request merged**